src/impl_build_cmd.rs

```diff
@@ -0,0 +1,267 @@
+use base64::{engine::general_purpose, Engine as _};
+
+use crate::RequestFlags;
+
+const MAX_KEY_LEN: usize = 250;
+const MAX_BIN_KEY_LEN: usize = 187; // 250 * 3 / 4 due to b64 encoding
+
+// type KeyHasher = Blake2b<U18>;
```

Comment:
Yes, will remove.

BTW I was doing blake2b hash before for long keys (and it was blazing fast), but decided against hardcoding this that could be something people want to do differently, and it is the rare case not worth optimizing. What I did move here is detecting if key is binary and doing the b64 since that is standard and uniform for all usages.